adapters/http-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -823,6 +823,31 @@ protected Integer getTimeUntilDisconnectFromRequest(final HttpContext ctx) {
         return ctx.getTimeTillDisconnect();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation limits the returned time period to 80 % of the configured idle timeout.
+     *
+     * @param tenant The tenant that the device belongs to.
+     * @param deviceTtd The TTD value provided by the device in seconds or {@code null} if the device did not
+     *                  provide a TTD value.
+     * @return A succeeded future that contains {@code null} if device TTD is {@code null}, or otherwise the lesser of
+     *         device TTD, the value returned by {@link TenantObject#getMaxTimeUntilDisconnect(String)} and 80% of the
+     *         configured idle timeout.
+     * @throws NullPointerException if tenant is {@code null}.
+     */
+    @Override
+    public Future<Integer> getTimeUntilDisconnect(final TenantObject tenant, final Integer deviceTtd) {
+        return super.getTimeUntilDisconnect(tenant, deviceTtd)
```

Comment:
FMPOV we should not invoke super.getTimeUntilDisconnect() here but instead copy the implementation to make it more robust against changes in the interface's default implementation.

Reply:
I've changed the method accordingly.